provenance/core.py

```diff
@@ -277,13 +315,65 @@ def wrapped(f):
         f._provenance_metadata = t.merge(kargs,
                                          {'version': version,
                                           'name': _name,
+                                          'archive_file': archive_file,
+                                          'delete_original_file': delete_original_file,
                                           'input_hash_fn': input_hash_fn,
                                           'input_process_fn': input_process_fn})
         f.__merge_defaults__ = merge_defaults
         return _provenance_wrapper(repo, f)
 
     return wrapped
 
+class ArchivedFile(object):
+    def __init__(self, id, original_filename=None, in_repo=True):
+        self.blob_id = id
+        self.original_filename = original_filename
+        self.in_repo = in_repo
+
+    def abspath(self):
+        repo = repos.get_default_repo()
+        path = repo.blobstore._filename(self.blob_id)
+        return os.path.abspath(path)
+
+    def __fspath__(self):
+        return self.abspath() if self.in_repo else self.original_filename
+
+    def __str__(self):
+        return self.__fspath__()
+
+    def __repr__(self):
+        return "ArchivedFile<{}>".format(self.blob_id)
```

Comment:
Right, I've fixed this and added some more information to the representation as well.